comp/api/api/apiimpl/internal/agent/agent.go

```diff
@@ -142,6 +145,45 @@ func getHostname(w http.ResponseWriter, r *http.Request) {
 	w.Write(j)
 }
 
+func configRequestHandler(w http.ResponseWriter, r *http.Request, cfg cfgcomp.Component) {
+	vars := mux.Vars(r)
+	subsection := vars["subsection"]
+
+	var body []byte
+	var err error
+
+	switch subsection {
+	case "ha":
+		settings := make(map[string]interface{})
+		settings["site"] = cfg.GetString("site")
+		settings["dd_url"] = cfg.GetString("dd_url")
+		settings["api_key"] = cfg.GetString("api_key")
```

Comment:
We don't scrub data here ? This would mean exposing sensitive information locally to any process.

We started working on something similar with #21025. The idea was to have a dedicated API for sharing sensitive information between agent processes. A different API allows us to build more restrictive security system in the future. It also allows customer to bind the CLI API on a different addr that the "sensitive API" (which is useful for container setups).

We might want to move the `ha` endpoint to this new API ?